include/zephyr/bluetooth/audio/hap.h

```diff
@@ -0,0 +1,394 @@
+struct bt_hap_harc_info {
+	/** Hearing Aid Type. */
+	enum bt_has_hearing_aid_type type;
+	/** Hearing Aid Features. */
+	enum bt_has_features features;
+	/** Hearing Aid Capabilities. */
+	enum bt_has_capabilities caps;
+	struct {
+		struct bt_hap_harc *pair;
+		const struct bt_csip_set_coordinator_csis_inst *csis;
+		const struct bt_csip_set_coordinator_set_member *member;
+	} binaural;
+};
+
+/** @brief Hearing Aid Remote Controller callback structure. */
+struct bt_hap_harc_cb {
+	/**
+	 * @brief Profile has been unbound.
+	 *
+	 * This callback notifies the application that the profile has been unbound.
+	 *
+	 * @param conn Controller instance pointer.
+	 */
+	void (*unbound)(struct bt_hap_harc *harc);
+
+	/**
+	 * @brief Controller has been connected.
+	 *
+	 * This callback notifies the application of a new connection to remote Hearing Aid device.
+	 * In case the err parameter is non-zero it means that the connection establishment failed.
+	 *
+	 * @param conn Controller instance pointer.
+	 * @param err 0 in case of success or negative errno or ATT Error code in case of error.
+	 */
+	void (*connected)(struct bt_hap_harc *harc, int err);
+
+	/**
+	 * @brief Controller has been disconnected.
+	 *
+	 * This callback notifies the application that the controller has been disconnected.
+	 *
+	 * @param conn Controller instance pointer.
+	 */
+	void (*disconnected)(struct bt_hap_harc *harc);
+
+	/**
+	 * @brief Controller info has been updated.
+	 *
+	 * This callback notifies the application that the controller info has been changed,
+	 * e.g. the Hearing Aid Features has been updated.
+	 *
+	 * @param conn Controller instance pointer.
+	 * @param info Info object.
+	 */
+	void (*info)(struct bt_hap_harc *harc, const struct bt_hap_harc_info *info);
+
+	sys_snode_t _node;
+};
+
+/**
+ * @brief Register Hearing Aid Remote Controller callbacks.
+ *
+ * @param cb The callback structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_hap_harc_cb_register(struct bt_hap_harc_cb *cb);
+
+/**
+ * @brief Unregister Hearing Aid Remote Controller callbacks.
+ *
+ * @param cb The callback structure.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_hap_harc_cb_unregister(struct bt_hap_harc_cb *cb);
+
+/**
+ * @brief Bind profile.
+ *
+ * Bind profile to remote device implementing HAP Hearing Aid role.
+ *
+ * @param conn Bluetooth connection object.
+ * @param[out] harc Valid controller object on success.
+ *
+ * @return Zero on success or (negative) error code on failure.
+ */
+int bt_hap_harc_bind(struct bt_conn *conn, struct bt_hap_harc **harc);
+
+/**
+ * @brief Unbind profile.
+ *
+ * Unbind profile or cancel pending profile binding.
+ *
+ * @param harc Controller instance pointer.
+ *
+ * @return Zero on success or (negative) error code on failure.
+ */
+int bt_hap_harc_unbind(struct bt_hap_harc *harc);
+
+/** @brief Hearing Aid Remote Controller Preset List callback structure. */
+struct bt_hap_harc_preset_cb {
```

Comment:
What is the purpose of having this as a separate struct from `struct bt_hap_harc_cb`?

Reply:
While multiple application modules might be interested in profile connection related callbacks (that's why I implemented `bt_hap_harc_cb` as callback list), `bt_hap_harc_preset_cb` the preset related callbacks, will be probably registered once by application module that caches (or not) the presets. The HAP HARC implementation does not define the size of cache, thus it's up to implementation whether it stores all of the presets read, or just a subset of it (that are specific to some scenario or vendor defined etc).